### ISSUE TITLE
Not an issue: But a question regarding "Forward" in Joint Embedding

### ISSUE THREAD

**Issue by BelaChakraborty** [technicolor-research/dsve-loc#12]
Hello,

I humbly request you, if you could kindly advice me: on lines 102 and 111, there is a "if condition" checking with "NONE"?

Can't directly it be passed without "if" checking?

And does this mean that both image and txt should be separately checked or there neednt be any checked.  Can I do something like:

           if imgs is not None or caps is not None:
               x_imgs = self.img_emb(imgs)
               x_imgs = self.dropout(x_imgs)
               x_imgs = self.fc(x_imgs)
     
              x_caps = self.cap_emb(caps, lengths=lengths)
              x_caps = x_caps / torch.norm(x_caps, 2, dim=1, keepdim=True).expand_as(x_caps)
        
              x_imgs = x_imgs / torch.norm(x_imgs, 2, dim=1, keepdim=True).expand_as(x_imgs)
              x_caps = x_caps / torch.norm(x_caps, 2, dim=1, keepdim=True).expand_as(x_caps)

          else: 
             x_imgs = None
             x_caps = None

Please if you could advice me, it would help me to analyse things better. 

Thank you
Bela

<img width="687" alt="Screenshot 2020-05-23 at 1 40 41 PM" src="https://user-images.githubusercontent.com/51686727/82720967-e02d2200-9cfb-11ea-8c52-c8aa1f94464c.png">

**Comment by M-Eng** [comment]
Hello,

Yes, you can do both check at once or no check at all.
It was coded with separated check in order to be able to use it to encode only texts (by passing a None image) or encode only images (by passing None text).

Best,
Martin

**Comment by BelaChakraborty** [comment]
Hello,

Thank you for your response. I had reproduced your code, but when I am running it after few month's again, I find a weird behaviour that the performance reduces after two epochs. When I re-produced it, seems like- it fails to work now.

I am trying to create a new enviroment, but I can install packages using your environmnent.yml.

Next, I don't find that pip install sru==2.1.3 support CUDA_FUNCTIONAL as i keep getting error. And if I download SRU from https://github.com/asappresearch/sru as suggested by one more person on GITHUB and do: pip install -r requirements.txt

I keep getting segmentation fault and it uninstall pytorch 1.0 which i installed using conda install pytorch==1.0.0 torchvision==0.2.1 cuda90 -c pytorch.

Could you advice on this issue?

Regards
Bela

________________________________

**Comment by M-Eng** [comment]
Hi, 
for the installation of the SRU you need to clone the right version 2.1.3
https://github.com/asappresearch/sru/tree/2.1.3

if you still have dependency problem try to install the requirement by hand one by one

**Comment by BelaChakraborty** [comment]
Hello

Thank you

Would look into it

regards
Bela
________________________________

**Comment by BelaChakraborty** [comment]
Hello,

 After you advised on the SRU unit, I ran the training, if you could see from the log, the image pipeline doesn't train at all for some reason. I couldn't reason out why? As I followed all the instructions given by you.

I guess you are the only one who could advice, what's causing this issue. I have downloaded the pre-trained imagenet weights and given its path in the config file.

I humbly request you if you could kindly advice on this issue.


thank you
regards
bela


[author_code.txt](https://github.com/technicolor-research/dsve-loc/files/4672837/author_code.txt)
